chia/full_node/mempool.py

```diff
@@ -155,6 +168,58 @@ def all_item_ids(self) -> List[bytes32]:
             cursor = self._db_conn.execute("SELECT name FROM tx")
             return [bytes32(row[0]) for row in cursor]
 
+    def transaction_ids_matching_coin_ids(self, coin_ids: Set[bytes32]) -> List[bytes32]:
+        transaction_ids: List[bytes32] = []
+
+        for transaction_id, item in self._items.items():
+            conds = item.npc_result.conds
+            if conds is None:
+                continue
+
+            for spend in conds.spends:
+                if spend.coin_id in coin_ids:
+                    transaction_ids.append(transaction_id)
+                    break
+
+                found_addition = False
+
+                for puzzle_hash, amount, _memo in spend.create_coin:
+                    if Coin(spend.coin_id, puzzle_hash, uint64(amount)).name() in coin_ids:
+                        transaction_ids.append(transaction_id)
+                        found_addition = True
+                        break
+
+                if found_addition:
+                    break
```

Comment:
```suggestion
                else:
                    continue
                break
```
Instead of using `found_additions` as state to know whether to exit the next level loop, you can use `for ... else`